
Initializing and Normalizing Rewards...
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/complete_moral_train.py", line 69, in <module>
    moral_train_n_experts(ratio, nb_experts, env, env_rad, lambd_str_list, ppo_filenames, discriminator_filenames, model_path, model_ext)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/moral_train_not_main.py", line 79, in moral_train_n_experts
    discriminator_list[-1].load_state_dict(torch.load(list_discrim_filenames[i], map_location=torch.device('cpu')))
  File "/home/lechapelier/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1497, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for Discriminator:
	Missing key(s) in state_dict: "reward_conv1.weight", "reward_conv1.bias", "reward_conv2.weight", "reward_conv2.bias", "reward_conv3.weight", "reward_conv3.bias", "value_conv1.weight", "value_conv1.bias", "value_conv2.weight", "value_conv2.bias", "value_conv3.weight", "value_conv3.bias".
	Unexpected key(s) in state_dict: "reward_l1.weight", "reward_l1.bias", "reward_l2.weight", "reward_l2.bias", "reward_l3.weight", "reward_l3.bias", "value_l1.weight", "value_l1.bias", "value_l2.weight", "value_l2.bias", "value_l3.weight", "value_l3.bias".
	size mismatch for reward_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
	size mismatch for value_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/complete_moral_train.py", line 69, in <module>
    moral_train_n_experts(ratio, nb_experts, env, env_rad, lambd_str_list, ppo_filenames, discriminator_filenames, model_path, model_ext)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/moral_train_not_main.py", line 79, in moral_train_n_experts
    discriminator_list[-1].load_state_dict(torch.load(list_discrim_filenames[i], map_location=torch.device('cpu')))
  File "/home/lechapelier/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1497, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for Discriminator:
	Missing key(s) in state_dict: "reward_conv1.weight", "reward_conv1.bias", "reward_conv2.weight", "reward_conv2.bias", "reward_conv3.weight", "reward_conv3.bias", "value_conv1.weight", "value_conv1.bias", "value_conv2.weight", "value_conv2.bias", "value_conv3.weight", "value_conv3.bias".
	Unexpected key(s) in state_dict: "reward_l1.weight", "reward_l1.bias", "reward_l2.weight", "reward_l2.bias", "reward_l3.weight", "reward_l3.bias", "value_l1.weight", "value_l1.bias", "value_l2.weight", "value_l2.bias", "value_l3.weight", "value_l3.bias".
	size mismatch for reward_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
	size mismatch for value_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/complete_moral_train.py", line 69, in <module>
    moral_train_n_experts(ratio, nb_experts, env, env_rad, lambd_str_list, ppo_filenames, discriminator_filenames, model_path, model_ext)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/moral_train_not_main.py", line 79, in moral_train_n_experts
    discriminator_list[-1].load_state_dict(torch.load(list_discrim_filenames[i], map_location=torch.device('cpu')))
  File "/home/lechapelier/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1497, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for Discriminator:
	Missing key(s) in state_dict: "reward_conv1.weight", "reward_conv1.bias", "reward_conv2.weight", "reward_conv2.bias", "reward_conv3.weight", "reward_conv3.bias", "value_conv1.weight", "value_conv1.bias", "value_conv2.weight", "value_conv2.bias", "value_conv3.weight", "value_conv3.bias".
	Unexpected key(s) in state_dict: "reward_l1.weight", "reward_l1.bias", "reward_l2.weight", "reward_l2.bias", "reward_l3.weight", "reward_l3.bias", "value_l1.weight", "value_l1.bias", "value_l2.weight", "value_l2.bias", "value_l3.weight", "value_l3.bias".
	size mismatch for reward_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
	size mismatch for value_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/complete_moral_train.py", line 69, in <module>
    moral_train_n_experts(ratio, nb_experts, env, env_rad, lambd_str_list, ppo_filenames, discriminator_filenames, model_path, model_ext)
  File "/home/lechapelier/Documents/Stage M2/Internship-Ethical-MORL/moral_rl/moral/moral_train_not_main.py", line 79, in moral_train_n_experts
    discriminator_list[-1].load_state_dict(torch.load(list_discrim_filenames[i], map_location=torch.device('cpu')))
  File "/home/lechapelier/.local/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1497, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for Discriminator:
	Missing key(s) in state_dict: "reward_conv1.weight", "reward_conv1.bias", "reward_conv2.weight", "reward_conv2.bias", "reward_conv3.weight", "reward_conv3.bias", "value_conv1.weight", "value_conv1.bias", "value_conv2.weight", "value_conv2.bias", "value_conv3.weight", "value_conv3.bias".
	Unexpected key(s) in state_dict: "reward_l1.weight", "reward_l1.bias", "reward_l2.weight", "reward_l2.bias", "reward_l3.weight", "reward_l3.bias", "value_l1.weight", "value_l1.bias", "value_l2.weight", "value_l2.bias", "value_l3.weight", "value_l3.bias".
	size mismatch for reward_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).
	size mismatch for value_out.weight: copying a param with shape torch.Size([1, 256]) from checkpoint, the shape in current model is torch.Size([1, 2704]).